src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long productId) {
+        return Storage.products.stream()
+                .filter(x -> x.getId().equals(productId))
```

Comment:
```suggestion
                .filter(p -> p.getId().equals(productId))
```
it's a good practice to use the first small letter of object: stream of strings -> s, Stream of Products -> p